src/main/java/uk/gov/ons/ctp/response/collection/exercise/client/impl/ActionSvcRestClientImpl.java

```diff
@@ -97,35 +109,62 @@ public ActionPlanDTO createActionPlan(
         collectionExerciseId,
         activeEnrolment);
 
-    MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
-    queryParams.add("collectionExerciseId", collectionExerciseId);
-    queryParams.add("activeEnrolment", activeEnrolment.toString());
-    UriComponents uriComponents =
-        restUtility.createUriComponents(appConfig.getActionSvc().getActionPlansPath(), queryParams);
-
-    ResponseEntity<List<ActionPlanDTO>> responseEntity;
-    try {
-      responseEntity =
-          restTemplate.exchange(
-              uriComponents.toString(),
-              HttpMethod.GET,
-              null,
-              new ParameterizedTypeReference<List<ActionPlanDTO>>() {});
-    } catch (HttpClientErrorException e) {
-      if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
-        return null;
-      }
-      throw e;
-    }
+    final MultiValueMap<String, String> queryParams = new LinkedMultiValueMap<>();
+    queryParams.add(SELECTOR_COLLECTION_EXERCISE_ID, collectionExerciseId);
+    queryParams.add(SELECTOR_ACTIVE_ENROLMENT, activeEnrolment.toString());
+    final ResponseEntity<List<ActionPlanDTO>> responseEntity =
+        getActionPlansByQueryParameters(queryParams);
 
     log.debug(
         "Successfully retrieved action plan for selectors, "
             + "collectionExerciseId: {}, activeEnrolment: {}",
         collectionExerciseId,
         activeEnrolment);
+
     return responseEntity.getBody();
   }
 
+  private ResponseEntity<List<ActionPlanDTO>> getActionPlansByQueryParameters(
+      final MultiValueMap<String, String> queryParams) {
+    final UriComponents uriComponents =
+        restUtility.createUriComponents(appConfig.getActionSvc().getActionPlansPath(), queryParams);
+
+    final ResponseEntity<List<ActionPlanDTO>> responseEntity =
+        restTemplate.exchange(
+            uriComponents.toString(),
+            HttpMethod.GET,
+            null,
+            new ParameterizedTypeReference<List<ActionPlanDTO>>() {});
+
+    return responseEntity;
+  }
+
+  @Override
+  public ActionPlanDTO getActionPlanBySelectors(
+      String collectionExerciseId, Boolean activeEnrolment) throws CTPException {
```

Comment:
any reason for using the Boolean class instead of the boolean primitive?

Reply:
getActionPlansBySelectors() uses a Boolean class which is where this is passed too. I can change both to use primitives? Why are primitives preferred over the class?